core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestion.java

```diff
@@ -253,6 +263,58 @@ protected XContentBuilder innerToXContent(XContentBuilder builder, Params params
                 return builder;
             }
 
+            private static ObjectParser<Map<String, Object>, Void> PARSER = new ObjectParser<>("CompletionOptionParser",
+                    true, HashMap::new);
+
+            static {
+                SearchHit.declareInnerHitsParseFields(PARSER);
+                PARSER.declareString((map, value) -> map.put(Suggestion.Entry.Option.TEXT.getPreferredName(), value),
+                        Suggestion.Entry.Option.TEXT);
+                PARSER.declareFloat((map, value) -> map.put(Suggestion.Entry.Option.SCORE.getPreferredName(), value),
+                        Suggestion.Entry.Option.SCORE);
+                PARSER.declareObject((map, value) -> map.put(CompletionSuggestion.Entry.Option.CONTEXT.getPreferredName(), value),
+                        Option::parseContext, CompletionSuggestion.Entry.Option.CONTEXT);
+            }
+
+            private static Map<String, Set<CharSequence>> parseContext(XContentParser parser, Void context) throws IOException {
```

Comment:
parseContexts? I think the names of the field and parsing method should reflect the name that is parsed, and in this case it's the plural form of context.